centreon/www/include/configuration/configGenerateTraps/formGenerateTraps.php

```diff
@@ -179,21 +183,27 @@
         if (isset($ret["apply"]) && $ret["apply"] && $returnVal == 0) {
             $msg_generate .= sprintf("<strong>%s</strong><br/>", _('Centcore commands'));
             foreach ($tab_server as $host) {
-                passthru("echo 'SYNCTRAP:" . $host['id'] . "' >> $centcore_pipe", $return);
+                passthru(
+                    escapeshellcmd("echo 'SYNCTRAP:{$host['id']}'") . ' >> ' . escapeshellcmd($centcore_pipe),
```

Comment:
You can use `escapeshellarg` when you have only one argument.
```suggestion
                    escapeshellcmd("echo 'SYNCTRAP:{$host['id']}'") . ' >> ' . escapeshellarg($centcore_pipe),
```